tests/integration/targets/test_move_objects_between_buckets/tasks/main.yml

```diff
@@ -0,0 +1,159 @@
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: '{{ security_token | default(omit) }}'
+      region: "{{ aws_region }}"
+
+  block:
+    # Define source bucket and destination bucket ==========================================
+    - name: Define bucket name used for tests
+      ansible.builtin.set_fact:
+        bucket:
+          src: "{{ bucket_name }}-src"
+          dest: "{{ bucket_name }}-dest"
+
+    # Create s3 buckets ====================================================================
+    - name: Create source s3 bucket
+      amazon.aws.s3_bucket:
+        name: "{{ bucket.src }}"
+        state: present
+
+    - name: Create destination s3 bucket
+      amazon.aws.s3_bucket:
+        name: "{{ bucket.dest }}"
+        state: present
+
+    # Put objects into s3 source buckets ===================================================
+    - name: Put object (text) in source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        object: /template/test.txt
+        content: "{{ lookup('file', 'test.txt') }}"
+        mode: put
+
+    - name: Put object (python) in source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        object: test.py
+        content: "{{ lookup('file', 'test.py') }}"
+        mode: put
+
+    - name: Obtaining image
+      ansible.builtin.slurp:
+        src: "{{ role_path }}/files/test.png"
+      register: put_binary
+
+    - name: Put object (image) in source s3 bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        object: put-binary.bin
+        content_base64: "{{ put_binary.content }}"
+        mode: put
+
+    # Move one object between buckets =======================================================
+    - name: Move one object between buckets
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.move_objects_between_buckets
+      vars:
+        move_objects_between_buckets_source_bucket: "{{ bucket.src }}"
+        move_objects_between_buckets_dest_bucket: "{{ bucket.dest }}"
+        move_objects_between_buckets_key_prefix: "template"
+
+    - name: Obtaining source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Check the source s3 bucket has two remaining objects
+      ansible.builtin.assert:
+        that:
+          - "bucket_src_info.s3_keys[0] == 'put-binary.bin'"
+          - "bucket_src_info.s3_keys[1] == 'test.py'"
+          - "bucket_src_info.s3_keys | length == 2"
+
+    - name: Obtaining destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.dest }}"
+        mode: list
+      register: bucket_dest_info
+
+    - name: Check the destination s3 bucket obtained one object
+      ansible.builtin.assert:
+        that:
+          - "bucket_dest_info.s3_keys[0] == 'template/test.txt'"
+          - "bucket_dest_info.s3_keys | length == 1"
+
```

Comment:
Should we add a test here to verify that attempting to delete the source bucket with objects still in it fails?

Reply:
I tried to add integration test for it, but when it runs this [task](https://github.com/redhat-cop/cloud.aws_ops/pull/77/files#diff-e4396f2b84325b7216ca34eeb1e3aaee12be962265c33add7e287d84e9106747R52), the playbook stops. For example:

```
    - name: Move one object between buckets and give an error trying to delete source bucket when source bucket is not empty
      ansible.builtin.include_role:
        name: cloud.aws_ops.move_objects_between_buckets
      vars:
        move_objects_between_buckets_source_bucket: "{{ bucket.src }}"
        move_objects_between_buckets_dest_bucket: "{{ bucket.dest }}"
        move_objects_between_buckets_key_prefix: "template"
        move_objects_between_buckets_delete_empty_source_bucket: true
      ignore_errors: true
```
The playbook stops when it reaches the [task](https://github.com/redhat-cop/cloud.aws_ops/pull/77/files#diff-e4396f2b84325b7216ca34eeb1e3aaee12be962265c33add7e287d84e9106747R52) although I putted `ignore_errors: true`.